### PR TITLE
Add a "server-per-process" mode

### PR DESCRIPTION
**Pull request by zamboney**
Kill the process by registering to the `end` event of `process.stdin` stream.

## TODO
- [x] add tests
- [x] open a fork for [this pr](https://github.com/ikseek/pipetests) with the changes i made: [done](https://github.com/ikseek/pipetests/pull/1)

### CODE REVIEW

File: js/packages/eyes-universal/src/cli.ts [hunk 1]
```diff
@@ -65,13 +65,27 @@ yargs
...
         },
       }),
     handler: async (args: any) => {
+      if (args['stdin-end-kill']) {
+        // register to `end` event of `stdin` stream
+        process.stdin.on('end', () => {
+          process.exit(0)
+        })
+        // need to trigger the data to catch the `end` event
+        process.stdin.on('data', () => {}) // eslint-disable-line @typescript-eslint/no-empty-function
```
Comment (kyryloonufriiev) [inline js/packages/eyes-universal/src/cli.ts:86]: Isn't calling `.resume()` is a more elegant way to switch streams in the flowing mode?

Comment (zamboney) [reply js/packages/eyes-universal/src/cli.ts:86]: i will test it in the `pipetests` repo, and if it's works i will change it

Comment (kyryloonufriiev) [reply js/packages/eyes-universal/src/cli.ts:86]: You can, but I'm most interested in having a test in this repo, which will show that the feature works.

Comment (zamboney) [reply js/packages/eyes-universal/src/cli.ts:86]: ok np

**Pull request merged**